Makefile

```diff
@@ -406,6 +408,41 @@ test/API.adoc: test/tools/libtest.sh
 %.pdf : %.xml
 	$(QUIET_DB2PDF)$(DOCBOOK2PDF) -o doc $<
 
+GENERATE_COMPILATION_DATABASE ?= no
+
+ifeq ($(GENERATE_COMPILATION_DATABASE),yes)
+compdb_check = $(shell $(CC) $(ALL_CFLAGS) \
+	-c -MJ /dev/null \
+	-x c /dev/null -o /dev/null 2>&1; \
+	echo $$?)
+ifneq ($(compdb_check),0)
+override GENERATE_COMPILATION_DATABASE = no
+$(warning GENERATE_COMPILATION_DATABASE is set to "yes", but your compiler does not \
+support generating compilation database entries)
+endif
+else
+ifneq ($(GENERATE_COMPILATION_DATABASE),no)
+$(error please set GENERATE_COMPILATION_DATABASE to "yes" or "no" \
+(not "$(GENERATE_COMPILATION_DATABASE)"))
+endif
+endif
+
+compdb_dir = compile_commands
+ifeq ($(GENERATE_COMPILATION_DATABASE),yes)
+compdb_file = $(compdb_dir)/$(subst /,-,$@.json)
+compdb_args = -MJ $(compdb_file)
+else
+compdb_args =
+endif
+
```

Comment:
Move this part at line 88 to make `$(missing_compdb_dir)` work.

Reply:
Yep, thanks! I'm not sure why because I don't see where the order would matter but that's okay :shrug: